### PR TITLE
Improve compact benchmark

### PR DESCRIPTION
**Pull request by marco6**
This PR aims at making the `compact` benchmark a little bit better.

The main problem with the current approach is that it measures insertions and deletions together with the compaction itself - and those two operations are taking the most of the time. This means that the result of the benchmark is not very representative of what compaction does.

Also, the test takes from [5](https://github.com/canonical/k8s-dqlite/actions/runs/9414094547/job/25932226910#step:5:11) to [7](https://github.com/canonical/k8s-dqlite/actions/runs/9414094547/job/25932226316#step:5:486) **minutes* to run a single iteration which is basically most of the execution time for the benchmark themselves. Running only one iteration makes the measure unreliable and difficult to compare.

This PR attempts to solve those issues by:
 - initializing the database with a single transaction, instead of one per item;
 - relating the size of the test to `b.N` so that Go can take measures accurately;
 - measuring time only during compaction;
 - switching the log off (to avoid corrupting the benchmark report).

It also moves the benchmark (and the test) to the `drivers` package as compaction is a backend operation (in fact, during tested logic, the `client` was used just to set up the test and as such has been removed).

Last, tests under kind folder were not running. Now they are, but I had to fix the `TestMigrate` as it wasn't working. If required that can be moved to another PR before this one.

### CODE REVIEW

**Comment by marco6** [comment]
Marking as draft since sqlite tests are not compiling if dqlite lib is missing.

**Review by bschimke95 — COMMENTED** [review]
Great work @marco6 

Did a first pass - code looks good to me. The PR does more than what it currently says, e.g.  updating the version command, adding dqlite tag fields etc. 

I would ask you to either put the rationale for that in the PR description or split this into a separate one.

**Comment by marco6** [comment]
Agreed. I will link here all the smaller PRs I create and mark this as draft for now.

**Comment by marco6** [comment]
Depends on #108 and #109

**Comment by marco6** [comment]
Depends on #110

File: pkg/kine/drivers/generic/generic.go [hunk 1]
```diff
@@ -72,6 +72,8 @@ var (
 		) AS high`
 )
 
+var RetryCount = 500
```
[the remark this answers is not in the document]
Comment (marco6) [reply pkg/kine/drivers/generic/generic.go:75]: This has been addressed in #110.

File: pkg/kine/logstructured/sqllog/sql.go [hunk 2]
```diff
@@ -12,6 +12,8 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+const SupersededCount = 100
```
[the remark this answers is not in the document]
Comment (marco6) [reply pkg/kine/logstructured/sqllog/sql.go:15]: This has been addressed in #111.

**Comment by marco6** [comment]
Depends on #111

File: pkg/kine/drivers/generic/generic.go [hunk 3]
```diff
@@ -474,9 +476,22 @@ func (d *Generic) GetCompactRevision(ctx context.Context) (int64, int64, error)
 		return 0, 0, fmt.Errorf("denied: %w", err)
 	}
 
-	row := d.DB.QueryRow(revisionIntervalSQL)
+	i := 0
```
[the remark this answers is not in the document]
Comment (marco6) [reply pkg/kine/drivers/generic/generic.go:479]: There was none. Addressed in #110

**Comment by marco6** [comment]
All pending PR were merged. Marking as ready.

File: pkg/kine/logstructured/sqllog/sql.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (marco6) [reply pkg/kine/logstructured/sqllog/sql.go:15]: At the end I had to keep it exported for now as it's needed in a test. In future rework we might think on making this unexported.

**Review by bschimke95 — APPROVED** [review]
LGTM, great work!

**Pull request merged**